feat: Support fetching accepted Activity Center notifications by multiple types

ilmotta opened the pull request:
## Summary

- [x] Changes endpoint `ActivityCenterNotificationsBy` to support fetching multiple types of notification in a single query.
- [x] Adds endpoint `UnreadAndAcceptedActivityCenterNotificationsCount` to allow the mobile client to fetch the count of unread & accepted notifications.
- [x] Add `golangci-lint` to Nix shell. This was possible since PR https://github.com/status-im/status-go/pull/3087 was merged.

#### Notes

- If you'd like to understand *why* these changes are needed, please see the [mobile PR](https://github.com/status-im/status-mobile/pull/14785), or issue https://github.com/status-im/status-mobile/issues/14712
- All changes should be completely backwards compatible, and there should be no impact for the desktop app.
- The mobile client has been already tested using this branch.

ilmotta committed:
Support fetching notifications by multiple types

ilmotta committed:
Make filtering by accepted field optional

ilmotta committed:
Use arrays in more places to be more consistent

ilmotta committed:
Test when array of notification types is empty

ilmotta committed:
Fix imports

ilmotta committed:
Add golangci-lint to Nix shell

ghost commented:
Hey @ilmotta, and thank you so much for making your first pull request in status-go! :heart: Please help us make your experience better by filling out this brief questionnaire https://goo.gl/forms/uWqNcVpVz7OIopXg2

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [x] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (12)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | cd32a642 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3088/1/) | 2023-01-14 13:12:55 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230114-131134-cd32a6-pr3088.zip) |
| :heavy_check_mark: | cd32a642 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3088/1/) | 2023-01-14 13:16:24 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230114-131134-cd32a6-pr3088.aar) |
| :heavy_check_mark: | cd32a642 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3088/1/) | 2023-01-14 13:18:30 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230114-131134-cd32a6-pr3088.zip) |
| :heavy_multiplication_x: | cd32a642 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/1/) | 2023-01-14 13:29:07 | ~17 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0e2249e5 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3088/2/) | 2023-01-19 22:46:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230119-224452-0e2249-pr3088.zip) |
| :heavy_check_mark: | 0e2249e5 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3088/2/) | 2023-01-19 22:49:01 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230119-224447-0e2249-pr3088.aar) |
| :heavy_check_mark: | 0e2249e5 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3088/2/) | 2023-01-19 22:49:36 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230119-224447-0e2249-pr3088.zip) |
| :heavy_check_mark: | 0e2249e5 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/2/) | 2023-01-19 22:55:12 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fa64eaa3 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3088/3/) | 2023-01-20 11:22:54 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230120-112151-fa64ea-pr3088.zip) |
| :heavy_multiplication_x: | fa64eaa3 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/3/) | 2023-01-20 11:23:08 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/3//consoleText) |
| :heavy_check_mark: | fa64eaa3 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3088/3/) | 2023-01-20 11:25:26 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230120-112146-fa64ea-pr3088.zip) |
| :heavy_check_mark: | fa64eaa3 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3088/3/) | 2023-01-20 11:25:29 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230120-112146-fa64ea-pr3088.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 5f544084 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/4/) | 2023-01-20 11:33:27 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/4//consoleText) |
| :heavy_check_mark: | 5f544084 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3088/4/) | 2023-01-20 11:34:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230120-113221-5f5440-pr3088.zip) |
| :heavy_check_mark: | 5f544084 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3088/4/) | 2023-01-20 11:35:32 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230120-113221-5f5440-pr3088.zip) |
| :heavy_check_mark: | 5f544084 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3088/4/) | 2023-01-20 11:36:49 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230120-113216-5f5440-pr3088.aar) |
| :heavy_multiplication_x: | 5f544084 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/5/) | 2023-01-20 12:00:00 | ~47 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e53b5d64 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3088/5/) | 2023-01-20 12:19:20 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230120-121812-e53b5d-pr3088.zip) |
| :heavy_check_mark: | e53b5d64 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3088/5/) | 2023-01-20 12:21:50 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230120-121812-e53b5d-pr3088.zip) |
| :heavy_check_mark: | e53b5d64 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3088/5/) | 2023-01-20 12:21:57 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230120-121812-e53b5d-pr3088.aar) |
| :heavy_check_mark: | e53b5d64 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/6/) | 2023-01-20 12:38:51 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3088/6//consoleText) |

MishkaRogachev commented on protocol/activity_center_persistence.go:
```diff
@@ -813,35 +816,53 @@ func (db sqlitePersistence) MarkActivityCenterNotificationsUnread(ids []types.He
-    ON
-    c.id = a.chat_id
-    WHERE NOT dismissed AND NOT a.accepted AND notification_type = ? AND author = ?`, ActivityCenterNotificationTypeContactRequest, contactID)
+		SELECT
```
Recently i was looking on those 3 selects.  It would be great to refactor common part from them

MishkaRogachev reviewed (approved):
<img width="558" alt="Screenshot 2023-01-16 at 12 46 50" src="https://user-images.githubusercontent.com/2522130/212640375-56fb540d-21c0-4964-a864-9363cecb20cc.png">

I've tested this changes with status-desktop, everything seems to be working

ilmotta commented:
> I've tested this changes with status-desktop, everything seems to be working

Thank you for double-checking @MishkaRogachev!

ilmotta committed:
Bump version

ilmotta committed:
Fix lint warnings

services/ens/api.go:77:2: field `settings` is unused (unused)
  settings *accounts.Database // NOTE: Accounts db is here for compatibility. Remove when settings.usernames is dropped.
  ^

protocol/messenger_communities.go:2010:3: SA2000: should call task.Waiter.Add(1) before starting the goroutine to avoid a race (staticcheck)
  task.Waiter.Add(1)
  ^
